### PR TITLE
kernel: Enable virtio-scsi

### LINKED ISSUE #201: Enable virtio-scsi in kernel configuration
We need to move to virtio-scsi from virtio-blk : https://github.com/containers/virtcontainers/issues/492

### PR DESCRIPTION
**Pull request by amshinde**
Need this config to move to virtio-scsi.

Fixes #201

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

**Review by egernst — APPROVED** [review]

File: kernel/kernel-config-4.9.x [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #
 # Automatically generated file; DO NOT EDIT.
-# Linux/x86 4.9.54 Kernel Configuration
+# Linux/x86 4.9.60 Kernel Configuration
```
Comment (jodh-intel) [inline kernel/kernel-config-4.9.x]: This change should be on a different commit I think?

File: kernel/kernel-config-4.9.x [hunk 2]
```diff
@@ -1241,9 +1241,85 @@ CONFIG_HAVE_IDE=y
 #
 CONFIG_SCSI_MOD=y
 # CONFIG_RAID_ATTRS is not set
-# CONFIG_SCSI is not set
-# CONFIG_SCSI_DMA is not set
+CONFIG_SCSI=y
+CONFIG_SCSI_DMA=y
```
Comment (grahamwhaley) [inline kernel/kernel-config-4.9.x:1245]: Can I ask - do we know if we need the SCSI_DMA set?
I was going to ask the same of the SG, but the virtio scsi driver does have SG code, and I can see how that could be used. Having DMA for a purely virtual device feels slightly odd though (but maybe it too is tied into how SG is passed/used?).

File: kernel/kernel-config-4.9.x [hunk 3]
```diff
@@ -603,7 +603,7 @@ CONFIG_PCI_MSI_IRQ_DOMAIN=y
 # CONFIG_HT_IRQ is not set
 CONFIG_PCI_ATS=y
 # CONFIG_PCI_IOV is not set
-# CONFIG_PCI_PRI is not set
+CONFIG_PCI_PRI=y
```
Comment (grahamwhaley) [inline kernel/kernel-config-4.9.x]: Do we know we need this? I looked it up - it allows PCI devices behind an iommu to recover from page faults - sounds a little 'fringe' for our virtio case?

File: kernel/kernel-config-4.9.x (hunk 1 above)
Comment (grahamwhaley) [inline kernel/kernel-config-4.9.x]: I agree with @jodh-intel - if you did this upgrade against a new kernel (and took a defconfig or allyes or similar), and/or you need to have the latest kernel for a feature, can you do the version upgrade as a separate commit (and possibly a separate PR?).

Comment (amshinde) [reply kernel/kernel-config-4.9.x]: We are currently using kernel 4.9.60, but our config file still contains 4.9.54!
I can put this change in a separate PR itself.

File: kernel/kernel-config-4.9.x [hunk 4]
```diff
@@ -1241,9 +1241,85 @@ CONFIG_HAVE_IDE=y
 #
 CONFIG_SCSI_MOD=y
 # CONFIG_RAID_ATTRS is not set
-# CONFIG_SCSI is not set
-# CONFIG_SCSI_DMA is not set
+CONFIG_SCSI=y
+CONFIG_SCSI_DMA=y
 # CONFIG_SCSI_NETLINK is not set
+# CONFIG_SCSI_MQ_DEFAULT is not set
```
Comment (mcastelino) [inline kernel/kernel-config-4.9.x]: @amshinde can we enable mq and see if we can leverage it in the virtio case? mq typically gives us clear performance and scaling benefits.

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/virtualization_tuning_and_optimization_guide/sect-virtualization_tuning_optimization_guide-blockio-techniques

Comment (amshinde) [reply kernel/kernel-config-4.9.x]: @mcastelino I have enabled this option

File: kernel/kernel-config-4.9.x (hunk 2 above)
Comment (amshinde) [reply kernel/kernel-config-4.9.x:1245]: @grahamwhaley This is enabled by default with virtio-scsi, maybe pulled in with SG as well.

File: kernel/kernel-config-4.9.x (hunk 3 above)
Comment (amshinde) [reply kernel/kernel-config-4.9.x]: @grahamwhaley Yes you are right, we dont typically need this for the virtio case, taking this out.

File: kernel/kernel-config-4.9.x (hunk 1 above)
Comment (jodh-intel) [reply kernel/kernel-config-4.9.x]: This sounds like a bit of a "process gap".

@grahamwhaley, @jcvenegas, @erick0z  - the packaging scripts run `make oldconfig`, but shouldn't we be committing the changes to avoid this sort of issue? Ideally, we'd build both `.deb` and `.rpm` packages, diff the two `make oldconfig`-updated `.config` files (they should be the same we'd hope!) then commit one of them as https://github.com/clearcontainers/packaging/blob/master/kernel/kernel-config-4.9.x so the comment line above is always in sync with the kernel version we're building?

**Comment by egernst** [comment]
I agree.  At the very least this should be done when the kernel is updated to a new dot release.

**Comment by amshinde** [comment]
@grahamwhaley Can you take another look?

**Comment by amshinde** [comment]
ping @grahamwhaley

**Review by grahamwhaley — APPROVED** [review]
lgtm
I know @egernst noted we could land this with a kernel minor upgrade - but I'm not sure how we'd co-ordinate that.
Merging...
